parsl/providers/azure/azure.py

```diff
@@ -0,0 +1,489 @@
+        for job_id in job_ids:
+            try:
+                vm = self.compute_client.virtual_machines.get(
+                    self.group_name, job_id, expand='instanceView')
+                status = vm.instance_view.statuses[1].display_status
+                statuses.append(translate_table.get(status, "UNKNOWN"))
+            # This only happens when it is in ProvisionState/Pending
+            except IndexError:
+                statuses.append("PENDING")
+        return statuses
+
+    def cancel(self, job_ids):
+        """Cancel the jobs specified by a list of job ids.
+        Parameters
+        ----------
+        job_ids : list of str
+            List of of job identifiers
+        Returns
+        -------
+        list of bool
+            Each entry in the list will contain False if the operation fails. Otherwise, the entry will be True.
+        """
+
+        return_vals = []
+
+        if self.linger:
+            logger.debug("Ignoring cancel requests due to linger mode")
+            return [False for x in job_ids]
+
+        for job_id in job_ids:
+            try:
+                logger.debug('\nDelete VM {}'.format(job_id))
+                async_vm_delete = self.compute_client.virtual_machines.delete(
+                    self.group_name, job_id)
+                async_vm_delete.wait()
+                self.instances.remove(job_id)
+                return_vals.append(True)
+            except Exception:
+                return_vals.append(False)
+
+        return return_vals
+
+    @property
+    def scaling_enabled(self):
+        return True
+
+    @property
+    def label(self):
+        return self._label
+
+    @property
+    def current_capacity(self):
+        """Returns the current blocksize."""
+        return len(self.instances)
+
+    def create_nic(self, network_client):
+        """Create (or update, if it exists already) a Network Interface for a VM.
+
+            Also ensures that there's a virtual network available.
+
+            We create a VPC with CIDR 10.0.0.0/16, which provides up to 64,000 instances.
+
+            We attach a subnet for each availability zone within the region specified in the
+            config. We give each subnet an ip range like 10.0.X.0/20, which is large enough
+            for approx. 4000 instances.
+
+
+        """
+        try:
+            logger.info('\nCreating (or updating) Vnet')
+            async_vnet_creation = self.network_client.virtual_networks.\
+                create_or_update(
+                    self.group_name, self.vnet_name, {
+                        'location': self.location,
+                        'address_space': {
+                            'address_prefixes': ['10.0.0.0/16']
+                        }
+                    })
+            vnet_info = async_vnet_creation.result()
+            self.resources["vnet"] = vnet_info
+
+        except Exception:
+            logger.info('Found Existing Vnet. Proceeding.')
```

Comment:
That exception text seems suspicious - what happens if the create fails for other reasons?

Reply:
Whoops - I meant to switch that over to a specific type of msrazure.CloudError. Thanks for catching that.

Reply:
Because of the way this is handled on the Azure end, my fix for this is very gross. If you have a better idea, please let me know and I will change it. 

The issue is that these errors don't actually appear to be subtypes of msrazure.CloudError - they seem to just be error messages in the string representations of the errors which are stylized as if they are actual python exceptions, eg `AzureError: InUseSubnetCannotBeDeleted`